internal/infrastructure/kubernetes/ratelimit_serviceaccount_test.go

```diff
@@ -0,0 +1,151 @@
+	assert.Equal(t, sa.Name, expectedProxyServiceAccountName(infra.Proxy.Name))
+
+	wantLabels := envoyAppLabel()
+	wantLabels[gatewayapi.OwningGatewayNamespaceLabel] = "default"
+	wantLabels[gatewayapi.OwningGatewayNameLabel] = infra.Proxy.Name
+	assert.True(t, apiequality.Semantic.DeepEqual(wantLabels, sa.Labels))
+}
+
+func TestCreateOrUpdateRateLimitServiceAccount(t *testing.T) {
+	testCases := []struct {
+		name    string
+		ns      string
+		in      *ir.RateLimitInfra
+		current *corev1.ServiceAccount
+		want    *corev1.ServiceAccount
+	}{
+		{
+			name: "create-ratelimit-sa",
+			ns:   "envoy-gateway-system",
+			in: &ir.RateLimitInfra{
+				Backend: &ir.RateLimitDBBackend{Redis: &ir.RateLimitRedis{URL: ""}},
+			},
+			want: &corev1.ServiceAccount{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ServiceAccount",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "envoy-gateway-system",
+					Name:      rateLimitInfraName,
+				},
+			},
+		},
+		{
+			name: "ratelimit-sa-exists",
+			ns:   "envoy-gateway-system",
+			in: &ir.RateLimitInfra{
+				Backend: &ir.RateLimitDBBackend{Redis: &ir.RateLimitRedis{URL: ""}},
+			},
+			want: &corev1.ServiceAccount{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ServiceAccount",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "envoy-gateway-system",
+					Name:      rateLimitInfraName,
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			kube := &Infra{
+				Namespace: tc.ns,
+			}
+			if tc.current != nil {
+				kube.Client = fakeclient.NewClientBuilder().WithScheme(envoygateway.GetScheme()).WithObjects(tc.current).Build()
+			} else {
+				kube.Client = fakeclient.NewClientBuilder().WithScheme(envoygateway.GetScheme()).Build()
+			}
+			err := kube.createOrUpdateRateLimitServiceAccount(context.Background(), tc.in)
+			require.NoError(t, err)
+
+			actual := &corev1.ServiceAccount{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: kube.Namespace,
+					Name:      rateLimitInfraName,
+				},
+			}
+			require.NoError(t, kube.Client.Get(context.Background(), client.ObjectKeyFromObject(actual), actual))
+
+			opts := cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion")
+			assert.Equal(t, true, cmp.Equal(tc.want, actual, opts))
+		})
+	}
+}
+
+func TestDeleteRateLimitServiceAccount(t *testing.T) {
+	testCases := []struct {
+		name   string
+		expect bool
+	}{
+		{
+			name:   "delete ratelimit service account",
+			expect: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			kube := &Infra{
+				Client:    fakeclient.NewClientBuilder().WithScheme(envoygateway.GetScheme()).Build(),
+				Namespace: "test",
+			}
+			rateLimitInfra := new(ir.RateLimitInfra)
+			err := kube.deleteRateLimitServiceAccount(context.Background(), rateLimitInfra)
```

Comment:
I dont see `expect` being used , and should there be another test to check if the SA resource was actually deleted from the kube cluster ?

Reply:
I will remove the `except`. and just this one place to check sa delete.